util/chunk/list.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chunk
+
+import (
+	"github.com/pingcap/tidb/types"
+)
+
+// List holds a slice of chunks, use to append rows with max chunk size properly handled.
+type List struct {
```

Comment:
In what scenario should we use this ?

Reply:
Sort and TopN can use it to simplify the logic.
And the IndexLookUpJoin need it too because we need index batch size greater than max chunk size.

Reply:
why not use `container/list` ?